### PR TITLE
[web-app] Sort by status, refactor a little more

### PR DESCRIPTION
**Pull request by styu**

### CODE REVIEW

File: packages/puzzle-utils/src/statusRank.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { IPuzzle } from "@superteamawesome/puzzle-api";
+
+export function statusRank(puzzle: Pick<IPuzzle, "status"> | undefined) {
+    if (puzzle === undefined) {
```
Comment (vsiao) [inline packages/puzzle-utils/src/statusRank.ts]: why would a puzzle be undefined :O

File: packages/web-app/app/src/components/puzzles/puzzleGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    tags,
+    onAddPuzzleToGroup,
+    onMarkAsMeta,
+    onPuzzleGroupDelete,
+    onPuzzleGroupSave,
+    onRemovePuzzleFromGroup,
+    onSavePuzzleChanges,
+}) => {
+    const [isPuzzleGroupCollapsed, setIsPuzzleGroupCollapsed] = React.useState<boolean>(
+        metaPuzzle?.status === "solved" || (metaPuzzle?.wasBacksolved ?? false),
```
Comment (vsiao) [inline packages/web-app/app/src/components/puzzles/puzzleGroup.tsx]: wasBacksolved is meaningless if status isn't `"solved"`

File: packages/web-app/app/src/components/puzzles/puzzleGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                        <tr>
+                            <th className="puzzle-name">Name</th>
+                            <th>Tags</th>
+                            <th>Created at</th>
+                            <th className="puzzle-links">Links</th>
+                            <th>Actions</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {sortedPuzzlesForGroup.map(([puzzleKey, puzzle]) => {
```
Comment (vsiao) [inline packages/web-app/app/src/components/puzzles/puzzleGroup.tsx]: I just read `key` directly from `puzzle` nowadays

File: packages/web-app/app/src/views/puzzles.tsx [hunk 4]
```diff
@@ -244,74 +222,35 @@ const PuzzlesView: React.FC = () => {
             <h3>Categorized puzzles</h3>
             <div className="puzzle-groups">
                 {Object.entries(puzzleGroups).map(([puzzleGroupKey, puzzleGroup]) => {
+                    const sortedPuzzlesForGroup: Array<[string | undefined, IPuzzle]> = Object.entries(
```
Comment (vsiao) [inline packages/web-app/app/src/views/puzzles.tsx]: think you just want `Object.keys` here

File: packages/puzzle-utils/src/statusRank.ts (hunk 1 above)
Comment (styu) [reply packages/puzzle-utils/src/statusRank.ts]: if you're like me and was testing the admin site and had bugs which leads to bad data..

File: packages/web-app/app/src/components/puzzles/puzzleGroup.tsx (hunk 2 above)
Comment (styu) [reply packages/web-app/app/src/components/puzzles/puzzleGroup.tsx]: heh, i guess it's just a redundant check then

File: packages/web-app/app/src/components/puzzles/puzzleGroup.tsx (hunk 3 above)
Comment (styu) [reply packages/web-app/app/src/components/puzzles/puzzleGroup.tsx]: sure, but i use the puzzleKey to write to the backend, so i want to be 100% sure im writing to the right firebase path

File: packages/web-app/app/src/views/puzzles.tsx (hunk 4 above)
Comment (styu) [reply packages/web-app/app/src/views/puzzles.tsx]: oops

**Pull request merged**